### PR TITLE
[AIRFLOW-2860] Update tests for druid hook

### PR DESCRIPTION
**Pull request by awelsh93**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-XXX
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This pull request updates the tests for `airflow/hooks/druid_hook.py` and also adds an assert so that `timeout` must be greater than or equal to 1 second. This should fix the failing build in #3707.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
I ran `nosetests /tests/hooks/test_druid_hook.py`

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

File: airflow/hooks/druid_hook.py [hunk 1]
```diff
@@ -53,6 +54,8 @@ def __init__(
         self.max_ingestion_time = max_ingestion_time
         self.header = {'content-type': 'application/json'}
 
+        assert self.timeout >= 1
```
Comment (feng-tao) [inline airflow/hooks/druid_hook.py]: I would prefer not to put assert outside of test code. Could you change it to throw exception when timeout is less than 1?

Comment (bolkedebruin) [reply airflow/hooks/druid_hook.py]: Also please use self.assertXXX otherwise we won't see causes.

**Comment by bolkedebruin** [comment]
I think this pr is blocking (as in it is required for) https://github.com/apache/incubator-airflow/pull/3708 I cannot get beyond the submit timeout test.

File: airflow/hooks/druid_hook.py (hunk 1 above)
Comment (bolkedebruin) [reply airflow/hooks/druid_hook.py]: Strike that, missed the location. I would handle this as a ValueError though

Comment (Fokko) [reply airflow/hooks/druid_hook.py]: @awelsh93 
Can you change this line to:
```
if self.timeout < 1:
   raise ValueError("Druid timeout should be equal or greater than 1")
```

Comment (awelsh93) [reply airflow/hooks/druid_hook.py]: Done - thanks

**Comment by awelsh93** [comment]
Thanks for merging this - can the change in #3707 get another review please? It was merged but then reverted due to tests failing which have now been fixed in this pull request.

**Pull request merged**